test/shopping_cart_test.rb

```diff
@@ -0,0 +1,32 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/product'
+require './lib/shopping_cart'
+
+class ShoppingCartTest < Minitest::Test
+  def test_it_exists
+    cart = ShoppingCart.new("King Soopers", "30items")
+    assert_instance_of ShoppingCart, cart
+  end
+
+  def test_it_has_readable_attributes
+    cart = ShoppingCart.new("King Soopers", "30items")
+    assert_equal "King Soopers", cart.name
+    assert_equal 30, cart.capacity
+    assert_equal [], cart.products
+  end
+
+  def test_can_add_products
+      cart = ShoppingCart.new("King Soopers", "30items")
+      product1 = Product.new(:paper, 'toilet paper', 3.70, '10')
+      product2 = Product.new(:meat, 'chicken', 4.50, '2')
+      product3 = Product.new(:paper, 'tissue paper', 1.25, '1')
+      product4 = Product.new(:produce, 'apples', 0.99, '20')
+      cart.add_product(product1)
+      cart.add_product(product2)
+      cart.add_product(product3)
+      assert_equal 13, cart.total_number_of_products
+      #assert_equal false, cart.is_full?
+      assert_equal [product1, product3], cart.products_by_category(:paper)
+  end
```

Comment:
This test seems to be verifying too many methods - it is named as if it is testing add_product (which it is testing indirectly), but it is also testing total_number_of_products and products_by_category.  We ideally would have three different tests for these three different methods.